lib/cc/services/github_pull_requests.rb

```diff
@@ -93,37 +102,44 @@ def add_comment
         body: COMMENT_BODY % @payload["compare_url"]
       }.to_json
 
-      http_post(comments_url, body)
+      post(comments_url, body) do |response|
+        doc = JSON.parse(response.body)
+        { id: doc["id"] }
+      end
     end
   end
 
   def receive_test_status
-    http_post(base_status_url("0" * 40), "{}")
-
-  rescue HTTPError => ex
-    if ex.status == 422 # response message: "No commit found for SHA"
-      { ok: true, message: "OAuth token is valid" }
-    else ex.status == 401 # response message: "Bad credentials"
-      { ok: false, message: ex.message }
+    url = base_status_url("0" * 40)
+    params = {}
+    raw_post(url, params.to_json)
+  rescue CC::Service::HTTPError => e
+    if e.status == 422
+      {
+        ok: true,
+        params: params.as_json,
+        status: e.status,
+        endpoint_url: url,
+        message: "OAuth token is valid"
+      }
+    else
+      raise
```

Comment:
Yes, it will. We only care about 422 here, as that's a "success". Anything else will be, as mentioned above, handled by the `WithErrorHandling` middleware.